File: week1/flipping-an-image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] A) {
        int rowSize = A.length;
        int colSize = A[0].length;
        // Reverse each row, A[i]
        for (int i = 0; i < rowSize; i++) {
            int lo = 0;
            int hi = rowSize - 1;
            while (lo < hi) {
                int temp = A[i][lo];
                A[i][lo] = A[i][hi];
                A[i][hi] = temp;
                lo++;
                hi--;
            }
        }
        // Invert each item in the matrix
        for (int i = 0; i < rowSize; i++) {
            for (int j = 0; j < colSize; j++) {
                A[i][j] ^= 1;
            }
        }
        return A;
    }
}

File: spring2020/unit1/flipping-an-image.py
class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        B = []
        for row in A:
            B.append(row[::-1])

        for i in range(len(B)):
            for j in range(len(B[0])):
                temp = B[i][j]
                B[i][j] = 1 - B[i][j]
        return B
